storage/src/main/scala/coop/rchain/storage/Storage.scala

```diff
@@ -0,0 +1,261 @@
+      lmdb.deleteKey(key.asInstanceOf[Key].toString, txn)
+    } else {
+      if (key.isInstanceOf[String]) {
+        try {
+          val k = new Key(key.asInstanceOf[String])
+          termTreeKeysSorted.remove(k)
+        } catch {
+          case e: Throwable => {}
+        }
+      }
+      lmdb.deleteKey(key, txn)
+    }
+  }
+
+  // delete a value associated with a key
+  def delete[K: Bbable, V: Bbable](
+      key: K,
+      value: V,
+      txn: Option[Txn[ByteBuffer]] = None): Boolean = {
+    lmdb.delete(key, value, txn)
+  }
+
+  // update a value associated with a key
+  def update[K: Bbable, V: Bbable](
+      key: K,
+      valueToBeReplaced: V,
+      valueReplaceWith: V,
+      txn: Option[Txn[ByteBuffer]] = None): Boolean = {
+    lmdb.update(key, valueToBeReplaced, valueReplaceWith, txn)
+  }
+
+  // return the size limitation imposed on keys by lmdb
+  def getMaxValueKeySize(): Int = {
+    lmdb.getMaxKeySize()
+  }
+
+  // Return the memory in use by the lmdb database, excluding
+  // other memory that supports the lmdb database
+  def memoryInUse(txn: Option[Txn[ByteBuffer]] = None): Long = {
+    lmdb.memoryInUse(txn)
+  }
+
+  def close(): Unit = {
+    lmdb.close()
+  }
+
+  // delete the data.mdb and lock.mdb files
+  def deleteFiles(): Unit = {
+    lmdb.deleteFiles()
+  }
+
+  protected[storage] def displayUniKeys: Unit = {
+
+    val keyIter = uniKeys
+    while (keyIter.hasNext()) {
+      val key = keyIter.next()
+      val keyStr = key.term
+      val valueStrs = lmdb.getStrings(keyStr)
+      print(keyStr + ": ")
+      for (value <- valueStrs.get) {
+        print(value + ", ")
+      }
+      println()
+    }
+  }
+
+  // Load a file of Keys and values to add to the database.
+  // Each line of the file must be of the form:
+  //     key value
+  // For example:
+  //     a(b(c(A), d(e(B))), f(C)) Y
+  def loadFile(filePath: String, display: Boolean = false): Unit = {
+    if (display) {
+      lmdb.Log(s"Load file: $filePath")
+      lmdb.Log("")
+    }
+
+    val source = Source.fromFile(filePath)
+    val lineIterator = source.getLines
+
+    for (lineOriginal <- lineIterator) {
+      val line = lineOriginal.trim
+
+      if (!line.isEmpty && line.slice(0, 2) != "//") {
+        val (key, value) = line.splitAt(line.lastIndexOf(' '))
+        val keyNoWhites = key.replaceAll("\\s", "")
+        val valueNoWhites = value.replaceAll("\\s", "")
+        put(keyNoWhites, valueNoWhites)
+      }
+    }
+  }
+}
+
+// StorageConfig is a class to configure a Storage object.
+// These parameters are explained in the Lmdb constructor.
+class StorageConfig {
+
+  var isKeyToValues: Boolean = false
+  var isWritable: Boolean = false
+  var baseDir: Option[String] = None
+  var dirName: Option[String] = None
+  var name: Option[String] = None
+  var maxSize: Long = Lmdb.minDbSize
+
+  def isValid(): Boolean = {
+
+    if (baseDir.isDefined && !new File(baseDir.get).isDirectory)
```

Comment:
[braces, returns](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards). Why not combine it into one big conditional instead of four separate ones?

Reply:
I thought the verbose version was more readable but I changed it.